lib/document_viewer_web/ensure_api_auth_token.ex

```diff
@@ -35,9 +35,12 @@ defmodule DocumentViewerWeb.EnsureApiAuthToken do
 
   @spec verify_token(String.t(), keyword()) :: :ok | :error
   defp verify_token(token, opts) do
-    api_auth_token = Keyword.get(opts, :api_auth_token)
+    api_auth_tokens =
+      opts
+      |> Keyword.get(:api_auth_tokens)
+      |> String.split(",")
 
-    if Plug.Crypto.secure_compare(token, api_auth_token) do
+    if Enum.any?(api_auth_tokens, &Plug.Crypto.secure_compare(token, &1)) do
```

Comment:
I'm not sure it matters, but to avoid timing differences depending on which possible token matched:
```elixir
Enum.reduce(api_auth_tokens, :error, fn to_compare, acc ->
  if Plug.Crypto.secure_compare(token, to_compare) do
    :ok
  else
    acc
  end
end)
```

Reply:
🤯